### PR TITLE
Game Engine v2.2 (new shuffleBoard method)

### PR DESCRIPTION
**Pull request by CSimbulan**
Rewrote the shuffle board method to be "better?"

Previous version:
-Used an array with 25 indices (9 blue, 8 red, 7 white, 1 black) for the colors
-Pick a random index for the color array, then assigned that value to a card, then removed that index from the array.

New version:
-Two arrays each with 4 indices. One with just the color names and another with the cumulative counts in the same order of the other array: ["blue", "red", "white", "black] and [9,17,24,25].
-Counter variable for many total cards are left to be assigned.
-Generate a random number between 0 and the amount of cards left.
-Use bisection to determine which indices the random number falls in between, then choose the upper bound e.g. if 18 is generated, since it is in between index 1 and 2, return 2. Use this index to get the color from the other array (in this example it is white.
-Subtract 1 from all the values in the counter array but only from the given index onward. E.g. using 18 again the array would become [9,17,23,24].
-If counts in index 0 hits 0 (all 9 blue cards are assigned) or if two adjacent indices are the same (e.g. we end up with [9,17,17,18] and in this case all white cards are assigned) then remove the index corresponding to the finished color in both the counts array and the color array.
-Decrement the total amount of cards left.

So there is no longer an ugly array with a bunch of the same words but it is now more complicated (more operations?).

Other minor changes:
-Moved words list to its own file, also added a few more words (other team members can add their own)

### CODE REVIEW

**Comment by dproc96** [comment]
As you mentioned, this is a pretty complicated solution to the problem. I wonder if you could simplify it by creating a function that takes an object structured like this
```javascript
{
  red: 8,
  blue: 9,
  black: 1,
  white: 7
}
```
and returns a shuffled array of those values like this `["red", "black", "white", "red", "red", "blue",... ]` that you can use to easily assign values. Maybe you can look into a more simple solution like that? I think it would also be more extendable if you wanted to add additional game modes (multiple black cards, more team cards, etc.)

**Comment by CSimbulan** [comment]
I kind of made a cleaner version of what I had originally with flexibility to change the color counts. I kept the assigning method the same as before because I'm not sure if its worth making the possible changes I thought of below:

The alternative way I thought of without making too many other changes was shuffle the color array (and maybe the words array too), then split off a chunk of 25 indices and do a for loop on one of them (just need 25 indices, or nested for loop of 5x5) and assign that way. If this is more efficient, or even if there's a better way I can still work on it.

File: server/utility/util.js [hunk 1]
```diff
@@ -22,7 +22,24 @@ const decodeJWT = (token) => {
   return decrypt.id;
 };
 
+const compare = (a, b) => {
+  return a < b ? -1 : a > b ? 1 : a >= b ? 0 : NaN;
+}
+
+const bisectLeft = (a, x) => {
```
Comment (bonnieli) [inline server/utility/util.js]: is this being used now?

Comment (CSimbulan) [reply server/utility/util.js]: Not anymore I'll remove them

**Review by bonnieli — APPROVED** [review]

**Pull request merged**